### PR TITLE
EOS-14652 S3 SW Update - Configs Merge

### PR DESCRIPTION
**Pull request by shalakadharap**

### CODE REVIEW

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/98/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/98/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/100/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/142/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/149/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/156/

File: rpms/s3/s3rpm.spec [hunk 1]
```diff
@@ -122,6 +122,20 @@ S3 server provides S3 REST API interface support for Motr object storage.
 %prep
 %setup -n %{name}-%{version}-%{_s3_git_ver}
 
+%pre
+if [ -f /opt/seagate/cortx/s3/conf/s3config.yaml.sample ]; then
```
Comment (nileshgovande) [inline rpms/s3/s3rpm.spec:126]: Make 2 layer wrapper script and lets remove intelligence from rpm post install

**Comment by nileshgovande** [comment]
Make unsafe attributes list instead of safe

File: scripts/swupdate/merge.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (shalakadharap) [reply scripts/swupdate/merge.sh:1]: fixed

File: scripts/swupdate/merge.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+then
+configValue=$(getProperty "$key" "$configFile" "$fileType")
+oldSampleValue=$(getProperty "$key" "$oldSampleFile" "$fileType")
+newSampleValue=$(getProperty "$key" "$newSampleFile" "$fileType")
+if [ "$configValue" = "$oldSampleValue" ] || [ -z "$configValue" ]
+then
+IFS=$'\n'
+if [ "$fileType" = "properties" ]
+then
+    sed -i "s/$key=$configValue/$key=$newSampleValue/g" $configFile
```
[the remark this answers is not in the document]
Comment (shalakadharap) [reply scripts/swupdate/merge.sh:49]: > log of this script

Fixed

**Comment by shalakadharap** [comment]
> Make unsafe attributes list instead of safe

Fixed

File: rpms/s3/s3rpm.spec (hunk 1 above)
Comment (shalakadharap) [reply rpms/s3/s3rpm.spec:126]: fixed

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/172/

**Comment by shalakadharap** [comment]
Duplicate and old

**Pull request closed without merging**